### PR TITLE
LANG-1542: ToStringBuilder.reflectionToString - Wrong JSON format for List of enums

### PR DESCRIPTION
**Pull request by TranNgocKhoa**
I add some code to fix an issue about **ToStringBuilder.reflectionToString - Wrong JSON format when object has a List of Enum**.
Link issue: https://issues.apache.org/jira/browse/LANG-1542

### CODE REVIEW

File: src/main/java/org/apache/commons/lang3/builder/ToStringStyle.java [hunk 1]
```diff
@@ -635,6 +635,14 @@ protected void appendDetail(final StringBuffer buffer, final String fieldName, f
      *  {@code toString}, not {@code null}
      */
     protected void appendDetail(final StringBuffer buffer, final String fieldName, final Collection<?> coll) {
+        if (coll != null && !coll.isEmpty()) {
+            coll.stream().findFirst()
+                    .map(Object::getClass)
+                    .filter(Class::isEnum)
```
Comment (swarajsaaj) [inline src/main/java/org/apache/commons/lang3/builder/ToStringStyle.java:641]: This will restrict only enums to be appended in a collection, a list of other types (e.g. Collection<Integer> or Collection<String>) will be ignored. This filter should be removed.

Comment (garydgregory) [reply src/main/java/org/apache/commons/lang3/builder/ToStringStyle.java:641]: We should avoid using streams here IMO, it will likely make performance worse in what can already be slow when reflection is involved.

Comment (TranNgocKhoa) [reply src/main/java/org/apache/commons/lang3/builder/ToStringStyle.java]: @garydgregory @swarajsaaj I added some code following your comments. Please take a look. Thank you!

Comment (swarajsaaj) [reply src/main/java/org/apache/commons/lang3/builder/ToStringStyle.java]: This should fix the case I mentioned earlier. Seems good.

Comment (TranNgocKhoa) [reply src/main/java/org/apache/commons/lang3/builder/ToStringStyle.java]: @swarajsaaj, I added test case for it

**Comment by kinow** [comment]
Will leave it to @garydgregory to do the final review as he had requested changes before :+1:

**Comment by kinow** [comment]
BTW, @TranNgocKhoa if you could rebase it, the build should pass. There were checkstyle issues on master until recently.

**Comment by XenoAmess** [comment]
Hi.
I don't think the changes in ToStringStyleTest.java is necessary...

File: src/test/java/org/apache/commons/lang3/builder/ToStringStyleTest.java [hunk 2]
```diff
@@ -16,10 +16,10 @@
  */
 package org.apache.commons.lang3.builder;
 
-import static org.junit.jupiter.api.Assertions.assertEquals;
-
 import org.junit.jupiter.api.Test;
 
+import static org.junit.jupiter.api.Assertions.assertEquals;
```
Comment (XenoAmess) [inline src/test/java/org/apache/commons/lang3/builder/ToStringStyleTest.java:21]: Hi.
I don't think the changes in ToStringStyleTest.java is necessary...

Comment (TranNgocKhoa) [reply src/test/java/org/apache/commons/lang3/builder/ToStringStyleTest.java]: @XenoAmess fixed, thank you!

**Comment by garydgregory** [comment]
-1: See below, I will provide an alternate implementation.
- The new method makes a copy of the collection as a new array, this can be done without the extra allocation.
- The arguments to assertEquals() are backward: The order is (expected, actual). If you do not follow the proper argument order, the failure messages are REALLY confusing ;-)
- The new tests do not cover edge cases: 
-- null array of enums
-- empty array of enums
-- null list of enums
-- empty list of enums
- New protected method is missing Javadoc since tag.
